pkg/controllers/datastoresyncer/datastoresyncer.go

```diff
@@ -103,6 +111,41 @@ func (d *DatastoreSyncer) Run(stopCh <-chan struct{}) error {
 		return err
 	}
 
+	if len(d.localCluster.Spec.GlobalCIDR) > 0 {
+		nodeName, ok := os.LookupEnv("NODE_NAME")
+		if !ok {
+			// Healthcheck in globalnet deployments will not work because of missing NODE_NAME.
+			klog.Error("Error reading the NODE_NAME from the env, healthChecker functionality will not work.")
+		} else {
+			d.localNodeName = nodeName
+			d.localFederator = syncer.GetLocalFederator()
+			resourceWatcher, err := watcher.New(&watcher.Config{
+				Scheme:     scheme.Scheme,
+				RestConfig: d.syncerConfig.LocalRestConfig,
+				ResourceConfigs: []watcher.ResourceConfig{
+					{
+						Name:                "Node watcher for datastoresyncer",
+						ResourceType:        &k8sv1.Node{},
+						ResourcesEquivalent: d.isNodeEquivalent,
+						Handler: watcher.EventHandlerFuncs{
+							OnCreateFunc: d.handleCreateOrUpdateNode,
+							OnUpdateFunc: d.handleCreateOrUpdateNode,
+							OnDeleteFunc: nil,
+						},
+					},
+				},
+			})
+			if err != nil {
+				return fmt.Errorf("error creating resource watcher %v", err)
+			}
+
+			err = resourceWatcher.Start(stopCh)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Can you move this into its own method, ie `startNodeWatcher`?

Reply:
Done